llvm/lib/Target/AMDGPU/AMDGPUAttributor.cpp

```diff
@@ -144,6 +144,20 @@ static bool funcRequiresHostcallPtr(const Function &F) {
 }
 
 namespace {
+
+std::optional<std::pair<unsigned, unsigned>> static parseRangeAttribute(
```

Comment:
Can you add an overload of getIntegerPairAttribute that takes the raw Attribute instead of duplicating the parsing? This also omits the context errors (I guess it's also bad that we'll spam them everywhere. We should probably make this an IR verifier check)